tests/src/texture/hipNormalizedFloatValueTex.cpp

```diff
@@ -55,14 +39,29 @@ __global__ void normalizedValTextureTest(unsigned int numElements, float* pDst)
     unsigned int elementID = hipThreadIdx_x;
     if(elementID >= numElements)
 	    return;
-    float coord =(float) elementID/(numElements-1);
+    float coord =(float) elementID/(numElements);
     pDst[elementID] = tex1D(textureNormalizedVal_1D, coord);
 }
 
+static float getNormalizedValue(const float value, const enum hipArray_Format texFormat) {
+  switch (texFormat) {
+    case HIP_AD_FORMAT_SIGNED_INT8:
+      return (value / SCHAR_MAX);
+    case HIP_AD_FORMAT_UNSIGNED_INT8:
+      return (value / UCHAR_MAX);
+    case HIP_AD_FORMAT_SIGNED_INT16:
+      return (value / SHRT_MAX);
+    case HIP_AD_FORMAT_UNSIGNED_INT16:
+      return (value / USHRT_MAX);
+    default:
+      return value;
+  }
+}
+
 template<typename T>
 bool textureTest(enum hipArray_Format texFormat)
 {
-    T hData[] = {65, 66, 67, 68, 69, 70, 71, 72,73,74};
+    T hData[] = {65, 66, 67, 68, 69, 70, 71, 72, 73, 74};
     T *dData = NULL;
     HIPCHECK(hipMalloc((void **) &dData, sizeof(T)*SIZE));
```

Comment:
Can you also update to use  hipMallocPitch to pass right pitch to hipTexRefSetAddress2D.

Reply:
Currently this test fails on hip-vdi because of this reason.